src/Widgets/AppContextMenu.vala

```diff
@@ -86,10 +89,28 @@ public class Slingshot.AppContextMenu : Gtk.Menu {
             add (plank_menuitem );
         }
 #endif
+        if (Environment.find_program_in_path ("io.elementary.appcenter") != null) {
+            if (!has_system_item && get_children ().length () > 0) {
+                add (new Gtk.SeparatorMenuItem ());
+            }
 
-        var appcenter = Backend.AppCenter.get_default ();
-        appcenter.notify["dbus"].connect (() => on_appcenter_dbus_changed.begin (appcenter));
-        on_appcenter_dbus_changed.begin (appcenter);
+            uninstall_menuitem = new Gtk.MenuItem.with_label (_("Uninstall")) {
+                sensitive = false
+            };
+            uninstall_menuitem.activate.connect (uninstall_menuitem_activate);
+
+            appcenter_menuitem = new Gtk.MenuItem.with_label (_("View in AppCenter")) {
```

Comment:
Would it be smarter to get the name of the app problematically so that we don't have to maintain translations here, as well? i.e.

```suggestion
            /// TRANSLATORS: string substitution is the name of the system package manager, e.g. AppCenter
            appcenter_menuitem = new Gtk.MenuItem.with_label (_("View in %s").printf (appstore_name)) {
```

See: sideload

https://github.com/elementary/sideload/blob/b6dbd3a45dbb3f55ec0ebf2b734bc4721656319b/src/Views/MainView.vala#L119-L137

https://github.com/elementary/sideload/blob/d9961a41f43e897964021201b81dc0fdd6176b26/src/Application.vala#L69-L76

Reply:
Since creating this menu item is conditional on the AppCenter binary existing and only sensitive when AppCenter dbus is available, it's unlikely that would ever return anything besides AppCenter

Reply:
oh derp you're right